kolibri/content/management/commands/importcontent.py

```diff
@@ -56,47 +56,54 @@ def add_arguments(self, parser):
         local_subparser = subparsers.add_parser(
             name='local',
             cmd=self,
-            help='Download the content from the given folder.'
+            help='Copy the content from the given folder.'
         )
+        local_subparser.add_argument('channel_id', type=str)
         local_subparser.add_argument('directory', type=str)
 
-    def handle_network_download(self, *args, **options):
+    def handle_transfer(self, method, *args, **options):
+
         channel_id = options["channel_id"]
 
         with using_content_database(channel_id):
             files = File.objects.all()
-            total_bytes_to_download = files.aggregate(Sum('file_size'))['file_size__sum']
+            total_bytes_to_transfer = files.aggregate(Sum('file_size'))['file_size__sum']
 
-            with self.start_progress(total=total_bytes_to_download) as overall_progress_update:
+            with self.start_progress(total=total_bytes_to_transfer) as overall_progress_update:
 
                 for f in files:
+
                     filename = f.get_filename()
-                    url = paths.get_content_storage_file_url(filename)
-                    path = paths.get_content_storage_file_path(filename)
+                    dest = paths.get_content_storage_file_path(filename)
 
                     # if the file already exists, add its size to our overall progress, and skip
                     # TODO(jamalex): could do md5 checks here instead, to be ultra-safe
-                    if os.path.isfile(path) and os.path.getsize(path) == f.file_size:
+                    if os.path.isfile(dest) and os.path.getsize(dest) == f.file_size:
                         overall_progress_update(f.file_size)
                         continue
 
-                    with transfer.FileDownload(url, path) as download:
+                    # determine where we're downloading/copying from, and create appropriate transfer object
+                    if method == "network":
+                        url = paths.get_content_storage_file_url(filename)
+                        filetransfer = transfer.FileDownload(url, dest)
+                    elif method == "local":
+                        srcpath = paths.get_content_storage_file_path(filename, datafolder=options["directory"])
+                        filetransfer = transfer.FileCopy(srcpath, dest)
```

Comment:
Not sure I fully understand the suggestion -- these strings are coming in from the command line parser (`kolibri manage importcontent -- *network* <channel_id>` vs `kolibri manage importcontent -- *local* <channel_id> <datapath>`), so I have to conditionalize on a string.

Oh, I see -- I was thinking this was the part inside the `handle_async` method. But I pass those strings into here, instead of having two separate functions, one for local and one for network. The reason I couldn't cleanly separate the logic (and remain DRY) is that I don't want to run lines [86-91](https://github.com/jamalex/kolibri/blob/4715c99f3c4067c0425ad7d45ffde81e88df5f75/kolibri/content/management/commands/importcontent.py#L86) until the check for line [81](https://github.com/jamalex/kolibri/blob/4715c99f3c4067c0425ad7d45ffde81e88df5f75/kolibri/content/management/commands/importcontent.py#L81) has passed (since creating the `FileDownload` object already creates a connection to the server, which needs to happen before we can do things like `filetransfer.total_size` on line [95](https://github.com/jamalex/kolibri/blob/4715c99f3c4067c0425ad7d45ffde81e88df5f75/kolibri/content/management/commands/importcontent.py#L95)).

So... I couldn't see a way to separate them without duplicating a large portion of the code from this method.